android/modules/network/src/java/ti/modules/titanium/network/socket/TCPProxy.java

```diff
@@ -220,25 +241,55 @@ public void run()
 			Object timeoutProperty = getProperty("timeout");
 
 			try {
-				if (timeoutProperty != null) {
-					int timeout = TiConvert.toInt(timeoutProperty, 0);
-
+				int port = TiConvert.toInt(getProperty("port"));
+				boolean useTls = TiConvert.toBoolean(getProperty("useTls"), false);
+
+				if (useTls) {
+					SocketFactory sslSocketFactory = new TiSocketFactory(null, null, NetworkModule.TLS_DEFAULT);
+					SSLSocket sslSocket = (SSLSocket) sslSocketFactory.createSocket();
+					sslSocket.setUseClientMode(true);
+					if (Build.VERSION.SDK_INT >= Build.VERSION_CODES.KITKAT) {
```

Comment:
`KITKAT` is API Level 19. This should be `VERSION_CODES.N` for API Level 24.
Personally, I would use the literal value `24` since there would be no chance of confusion and saves time looking it up if you don't have this memorized. I'm not opposed to using Google's version constants, but I find using literal API Level values helps avoid mistakes.

The SNI code below is to check for man-in-the-middle attacks, right? So, it's okay to run without it on older OS versions. Our [appcelerator.https](https://github.com/appcelerator-modules/appcelerator.https) module can be used in conjunction with `HTTPClient` to check for man-in-the-middle attacks. Perhaps the solution in the future is to extend our `TCP` object to support our module's returned `SecurityManager` in the future.

Reply:
Oh, damn! Must have looked up the wrong version code. I thought using version codes would be more expressive, but my own mistake just proved me wrong i guess.

The SNI code is to support servers that serve multiple domains via vhosts. With SNI the server will provide the correct certificate for the requested host. The alternative for older devices would be to provide a way to manually verify the certificate host name, see https://developer.android.com/training/articles/security-ssl#WarningsSslSocket. I've added the `validateServerIdentity` option that allows a user to manually verify the server's identity in such cases. It's modeled after node's [tls.checkServerIdentity](https://nodejs.org/api/tls.html#tls_tls_checkserveridentity_hostname_cert)

For more sophisticated checks or SSL pinning our https module might indeed come handy. But yeah, that's something for the future.